File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let map = new Map();
    
    for (let i=0; i<participant.length; i++) {
        let name = participant[i];
        map[name] = (map[name] || 0) + 1;
    }
    
    for (let i=0; i<completion.length; i++) {
        let name = completion[i];
        map[name] -= 1;
    }
    
    for (let name in map) {
        if (map[name] > 0) return name
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic = { }
    
    for p in participant:
        if p in dic:
            dic[p] += 1
        else:
            dic[p] = 1
    
    for c in completion:
        dic[c] -= 1
        
    for key in dic.keys():
        if dic[key] > 0:
            return key
